topdown/topdown_test.go

```diff
@@ -95,6 +95,50 @@ func TestTopDownQueryIDsUnique(t *testing.T) {
 	}
 }
 
+func TestTopDownIndexExpr(t *testing.T) {
+	ctx := context.Background()
+	store := inmem.New()
+	inputTerm := &ast.Term{}
+	txn := storage.NewTransactionOrDie(ctx, store)
+	defer store.Abort(ctx, txn)
+
+	compiler := compileModules([]string{
+		`package test
+
+		p = true {
+		     1 > 0
+		     q
+		}
+
+		q = true { true }`})
+
+	tr := []*Event{}
+
+	query := NewQuery(ast.MustParseBody("data.test.p")).
+		WithCompiler(compiler).
+		WithStore(store).
+		WithTransaction(txn).
+		WithTracer((*BufferTracer)(&tr)).
+		WithInput(inputTerm)
+
+	_, err := query.Run(ctx)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	var expectedQueryID uint64
```

Comment:
It would be better if the test case asserted that the index events in the trace contain the expected AST node. With the current test, we're only asserting that the query ID is correct (which does not exercise the fix.) I would probably do something like...

```golang
exp := []*ast.Expr{
  ast.MustParseExpr("data.test.p")),
  ast.MustParseExpr("data.test.q")),
}

i := 0 

for _, evt := range tr {

  if evt.Op != IndexOp {
    continue
  }

  expr, ok := evt.Node.(*ast.Expr)
  if !ok {
    t.Fatal("Expected expr node but got:", evt.Node)
  }

  if ast.Compare(expr, exp[i]) != 0 {
    t.Fatalf("Expected %v but got: %v", exp[i], expr)
  }

  i++
}
```

Reply:
Updated test